pinecone_text/dense/cohere_encoder.py

```diff
@@ -0,0 +1,84 @@
+from enum import Enum
+from typing import Union, List, Any, Optional
+from pinecone_text.dense.base_dense_ecoder import BaseDenseEncoder
+
+try:
+    import cohere
+    from cohere.error import CohereError
+except (OSError, ImportError, ModuleNotFoundError) as e:
+    _cohere_installed = False
+else:
+    _cohere_installed = True
+
+
+class CohereInputType(Enum):
+    SEARCH_DOCUMENT = "search_document"
+    SEARCH_QUERY = "search_query"
+    # CLASSIFICATION = "classification"
+    # CLUSTERING = "clustering"
+
+
+class CohereEncoder(BaseDenseEncoder):
```

Comment:
Can you please add unit and system tests same as we have for OpenAI embeddings? I added a repository secret for Cohere under `COHERE_API_KEY`. You can use it in the github action running the system tests same as for OpenAI